titus-server-master/src/main/java/com/netflix/titus/master/scheduler/TaskPlacementFailure.java

```diff
@@ -79,31 +79,46 @@
          */
         OpportunisticResource,
 
+        /**
+         * Task not launched due to other tasks with the exclusiveHost hard constraint on the same agent. It has the lowest priority.
+         */
+        ExclusiveHost,
```

Comment:
Is the priority defined by the declaration order in the enum type?

Reply:
No, the priority is defined by how `TaskPlacementFailureClassifier` uses each `FailureKind` defined here. I will add more comments to reflect that.